src/dbnode/storage/namespace.go

```diff
@@ -750,6 +750,37 @@ func (n *dbNamespace) AggregateQuery(
 	return res, err
 }
 
+func (n *dbNamespace) PrepareBootstrap() ([]databaseShard, error) {
+	var (
+		wg           sync.WaitGroup
+		multiErrLock sync.Mutex
+		multiErr     xerrors.MultiError
+		shards       = n.GetOwnedShards()
+	)
+	for _, shard := range shards {
+		shard := shard
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			err := shard.PrepareBootstrap()
+			if err != nil {
+				multiErrLock.Lock()
+				multiErr = multiErr.Add(err)
```

Comment:
May be worth logging here to make bootstrap debugging a little easier?

Reply:
I do that below with I believe with bootstrap prepare failed?  I can wrap the error potentially.